src/examples/go_monorepo_workflows.yml

```diff
@@ -0,0 +1,187 @@
+            context: dev
+            name: deploy-master-dev
+            work_dir: './api'
+            requires:
+              - docker-master
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: prod
+            context: prod
+            name: deploy-prod
+            work_dir: './api'
+            requires:
+              - deploy-master-dev
+
+    worker:
+      when:
+        or:
+          - << pipeline.parameters.worker >>
+          - << pipeline.parameters.common >>
+      jobs:
+        - quality-gate:
+            context: dev
+            appname: worker
+        - request_branch_deploy_dev:
+            type: approval
+            name: request-deploy
+            filters:
+              branches:
+                ignore:
+                  - master
+        - ric-orb/build_push_image:
+            <<: *cfg-isopod-version
+            context: dev
+            work_dir: './worker'
+            name: docker-branch
+            requires:
+              - request-deploy
+              - quality-gate
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: dev
+            context: dev
+            name: deploy-dev
+            work_dir: './worker'
+            requires:
+              - docker-branch
+        #master branch
+        - ric-orb/build_push_image:
+            <<: *cfg-isopod-version
+            context: prod
+            prebuild_steps:
+              - attach_workspace:
+                  at: .
+            work_dir: './worker'
+            name: docker-master
+            filters:
+              branches:
+                only:
+                  - master
+            requires:
+              - quality-gate
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: dev
+            context: dev
+            name: deploy-master-dev
+            work_dir: './worker'
+            requires:
+              - docker-master
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: prod
+            context: prod
+            name: deploy-prod
+            work_dir: './worker'
+            requires:
+              - deploy-master-dev
+
+    no-code-change:
+      when:
+        and:
+          - not: << pipeline.parameters.api >>
+          - not: << pipeline.parameters.worker >>
+          - not: << pipeline.parameters.common >>
+      jobs:
+        - ric-orb/no_op
+
+  jobs:
+    quality-gate:
+      parameters:
+        appname:
+          type: string
+      executor:
+        name: go/default
+        <<: *cfg-go-version
+      steps:
+        - checkout
+        - ric-orb/install_build_test:
+            work_dir: << parameters.appname >>
+        - run:
+            name: Running tests on common
+            command: make test
+            working_directory: common
+        - persist_to_workspace:
+            root: .
+            paths:
+              - ./<< parameters.appname >>/app
```

Comment:
this could be a go specific job similar to `maven_build_test` for java
if we do that we just have to make sure that it also works with non-monorepos

Reply:
i'll try to do it